openml-lightgbm/openml-lightgbm-provider/pom.xml

```diff
@@ -0,0 +1,180 @@
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.feedzai</groupId>
+            <artifactId>openml-utils</artifactId>
+            <type>test-jar</type>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.assertj</groupId>
+            <artifactId>assertj-core</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-csv</artifactId>
+            <scope>test</scope>
+        </dependency>
+
+    </dependencies>
+
+    <build>
+
+        <resources>
+            <resource>
+                <directory>${project.basedir}/target/resources</directory>
+            </resource>
+        </resources>
+
+        <plugins>
+
+            <!-- Copy the LightGBM shared libraries to the resources folder -->
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-resources-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>copy-shared-libs-resources</id>
+                        <phase>generate-resources</phase>
+                        <goals>
+                            <goal>copy-resources</goal>
+                        </goals>
+                        <configuration>
+                            <outputDirectory>${project.basedir}/target/resources</outputDirectory>
+                            <resources>
+                                <resource>
+                                    <directory>../lightgbm-builder/lightgbmlib_build</directory>
+                                    <include>*.so</include>
+                                    <include>*.so.1.0.0</include>
+                                </resource>
+                            </resources>
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <id>copy-licences</id>
+                        <phase>generate-resources</phase>
+                        <goals>
+                          <goal>copy-resources</goal>
+                        </goals>
+                        <configuration>
+                            <outputDirectory>${project.basedir}/target/resources</outputDirectory>
+                            <resources>
+                                <resource>
+                                    <directory>LICENCES</directory>
+                                </resource>
+                            </resources>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+
+            <!-- assemble package -->
+            <plugin>
```

Comment:
Out of curiosity: why to you need this? To merge the extra jar into this one?

Reply:
Ideally we should need to carry around only a single jar which has lightgbmlib and the .so's inside.

I think this is needed for that? @JPDSousa 

What I'd like to see is the parent pom assemble a jar with this jar + lightgbmlib so we match the conventions in the openml-feedzai-java for the other providers (as currently the root README for lightgbm is wrong).